### PR TITLE
Bump Optimum to 1.22 + Adapt to the SD task refactoring in Optimum main

### PR DESCRIPTION
**Pull request by JingyaHuang**
# What does this PR do?

Make optimum-neuron compatible with changes in https://github.com/huggingface/optimum/pull/1947. Besides, with the changes, the `InstructPix2Pix` support in Optimum Neuron will fail, a PR is opened in Optimum repo to patch this: https://github.com/huggingface/optimum/pull/1999. 

Merge this only when we bump to a stable Optimum release including the two PRs mentioned above.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: tests/inference/test_modeling.py [hunk 1]
```diff
@@ -13,1884 +13,1512 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import gc
-import os
```
Comment (dacorvo) [inline tests/inference/test_modeling.py:16]: Why so many lines were removed and other commented out ?

**Review by dacorvo — COMMENTED** [review]
Thanks for the pull-request. It looks good to me except for the changes in test_modleing.py that I did not understand.

File: tests/inference/test_modeling.py (hunk 1 above)
Comment (JingyaHuang) [reply tests/inference/test_modeling.py:16]: Good catch, I accidentally remove them for debugging a specific test.... Putting it back.

**Review by dacorvo — APPROVED** [review]
LGTM, thank you for the pull-request !

**Pull request merged**